charts/argo-cd/values.yaml

```diff
@@ -1007,6 +1007,9 @@ server:
     duration: ""
     # -- How long before the currently issued certificate's expiry cert-manager should renew the certificate. Value must be in units accepted by Go time.ParseDuration
     renewBefore: ""
+    # -- The private key algorithm to use in the Certificate spec.
+    privateKey:
+      algorithm: ""
```

Comment:
How about exposing whole [CertificatePrivateKey](https://cert-manager.io/docs/reference/api-docs/#cert-manager.io/v1.CertificatePrivateKey) spec like below with sensible defaults?

```yaml
# Private key of the certificate
privateKey:
  # -- Rotation policy of private key when certificate is re-issued. Either: `Never` or `Always`
  rotationPolicy: Never
  # -- The private key cryptography standards (PKCS) encoding for private key. Either: `PCKS1` or `PKCS8`
  encoding: PKCS1
  # -- Algorithm used to generate certificate private key. One of: `RSA`, `Ed25519` or `ECDSA`
  algorithm: RSA
  # -- Key bit size of the private key. If algorithm is set to `Ed25519`, size is ignored.
  size: 2048
```

Reply:
This has been updated with your recommendations.  Good idea.